pkgs/development/libraries/mesa/stubs.nix

```diff
@@ -1,50 +1,37 @@
 { stdenv
+, lib
 , libglvnd
-, mesa
 , OpenGL
 , testers
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "libGL";
-  inherit (if stdenv.hostPlatform.isDarwin then mesa else libglvnd) version;
+  version = if stdenv.hostPlatform.isDarwin then "4.1" else libglvnd.version;
   outputs = [ "out" "dev" ];
 
-  # On macOS, libglvnd is not supported, so we just use what mesa
-  # build. We need to also include OpenGL.framework, and some
-  # extra tricks to go along with. We add mesa’s libGLX to support
-  # the X extensions to OpenGL.
+  # On macOS, libglvnd is not supported, and mesa no longer builds (but it only
+  # provided a software renderer anyway). Provide the OpenGL framework as well
+  # as a pkg-config file for OpenGL.framework.
+  # GLX is not supported nor is OpenGL ES.
   buildCommand = if stdenv.hostPlatform.isDarwin then ''
-    mkdir -p $out/nix-support $dev
-    echo ${OpenGL} >> $out/nix-support/propagated-build-inputs
-    ln -s ${mesa.out}/lib $out/lib
+    mkdir -p $out/nix-support $dev/lib/pkgconfig $dev/nix-support
 
-    mkdir -p $dev/lib/pkgconfig $dev/nix-support
+    echo ${OpenGL} >> $out/nix-support/propagated-build-inputs
```

Comment:
This won’t work with `overrideSDK`. The OpenGL framework needs to be added to `propagatedBuildInputs`, so `overrideSDK` can know about the framework dependency being propagated.